doc/developer/design/20210831_correctness.md

```diff
@@ -0,0 +1,124 @@
+# <Insert Name of Feature>
+
+## Summary
+
+A description of the correctness properties Materialize provides and the terms and concepts we use to describe them.
+
+## Goals
+
+The main goal is to lay the foundation for ongoing work to clearly articulate correctness properties of Materialize.
+This includes the introduction of concepts that underlie correctess, the components that will enforce them, and how they can communicate clearly.
+
+## Non-Goals
+
+It is not a goal to prescribe fine detail in how each of the components should behave, or which should have certain responsibilities.
+It is not a goal to prescribe which behavior Materialize should support, but only how to speak about a "correct" subset of it.
+It is not a goal to advocate (much) for DEFINITE over NON-VOLATILE.
+
+## Description
+
+Materialize means to provide several correctness guarantees, which manifest as various consistency and durability properties.
+At the heart of this is the idea that the ordering of events can be explicitly resolved and recorded, after which queries and views over the data become a matter of correct evaluation on completed subsets of time.
+The guarantees of Materialize derive from the durability of these recorded data, and the determinism of views over them.
+
+The primary term introduced here is "definite" to modify a collection of data.
+A collection is definite if it is uniquely named or defined such that anyone with the same identification will see the same collection of data at the same logical times.
+A definite collection has had its updates put in a specific order, by way of the timestamps on updates.
```

Comment:
"by way of _durable_ timestamps"?

Reply:
Yes, though all I really meant here was "order comes by way of the timestamps" where durability is orthogonal. Perhaps instead it should be "specific, durable order" and then "durable timestamps", but the intent was "you get order from timestamps" leaving "if the timestamps are durable, so is the order, and if not then not" unsaid. But it can be said!

Reply:
It should probably be said. I think the implication is clear to you and me, but might not be to fresh readers.